Packs/CrowdStrikeFalcon/TestPlaybooks/Test_Playbook_-_CrowdStrike_Falcon_-_Get_Detections_by_Incident.yml

```diff
@@ -0,0 +1,4245 @@
+      version: -1
+      name: Verify CrowdStrike Found Detections
+      description: Verify that the 'FoundDetections' context key was populated and its value was set to 'false'.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "227"
+      Verified:
+      - "264"
+    separatecontext: false
+    conditions:
+    - label: Verified
+      condition:
+      - - operator: isEqualString
+          left:
+            value:
+              complex:
+                root: CrowdStrike
+                accessor: FoundDetections
+            iscontext: true
+          right:
+            value:
+              simple: "False"
+          ignorecase: true
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": -180,
+          "y": -1600
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "246":
+    id: "246"
+    taskid: 5f5d096e-c5ce-4fec-8948-2d08b9f6f3d1
+    type: condition
+    task:
+      id: 5f5d096e-c5ce-4fec-8948-2d08b9f6f3d1
+      version: -1
+      name: Verify Process Start Time
+      description: Verify that the 'CrowdStrike.Detection.ProcessStartTime' context key was populated.
+      type: condition
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#default#':
+      - "254"
+      Verified:
+      - "269"
+    separatecontext: false
+    conditions:
+    - label: Verified
+      condition:
+      - - operator: isNotEmpty
+          left:
+            value:
+              complex:
+                root: CrowdStrike.Detection
+                accessor: ProcessStartTime
+            iscontext: true
+          ignorecase: true
+    continueonerrortype: ""
+    view: |-
+      {
+        "position": {
+          "x": -170,
+          "y": -420
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "248":
+    id: "248"
+    taskid: 4ab40d4c-f6c1-4ac9-826d-e89721534ed2
+    type: playbook
+    task:
+      id: 4ab40d4c-f6c1-4ac9-826d-e89721534ed2
+      version: -1
+      name: CrowdStrike Falcon - Get Detections by Incident
+      description: |-
+        This playbook is part of the 'Malware Investigation And Response' pack. For more information, refer to https://xsoar.pan.dev/docs/reference/packs/malware-investigation-and-response.
+        This playbook enables getting CrowdStrike Falcon detection details based on the CrowdStrike incident ID.
+      playbookName: CrowdStrike Falcon - Get Detections by Incident
+      type: playbook
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "361"
+    scriptarguments:
+      IncidentID:
+        simple: inc:046761c46ec84f40b27b6f79ce7cd32c:b249f531b46849ecb576bf429cc4cc78
```

Comment:
I think we should do here another conditional task to validate if this incident is alive. Sometimes our NDR evaluations might delete/rotate data (like every month or 6). So in that case the PB should take another available inc.

Reply:
The following changes were made to the playbook in order to validate CrowdStrike Falcon live incidents:
1- Added a new task ( task number 363) that uses the 'cs-falcon-list-incident-summaries' automation to obtains a list of CrowdStrike Falcon incidents before performing the tests.

2- To obtain CrowdStrike Falcon incident IDs, I configured the 'IncidentID' input of the 'CrowdStrike Falcon - Get Detections by Incident' sub-playbook by using the first array element of the context key 'CrowdStrike.Incidents.incident_id' generated by the automation in the previous task.